### DISCUSSION TITLE
Change m141

### DISCUSSION THREAD

**Discussion by bluemorphe** [jschuh/klipper-macros#195] [Q&A] [answered]
I would like to modify the gcode m141 so that it interacts with my fan but it is blocked in read only I already have the macro

https://docs.vorondesign.com/community/howto/alchemyEngine/chamber_temperature_exhaust_fan.html

**Comment by jschuh** [comment, marked as answer]
You just need to override m141 with something like this:

```
[gcode_macro M141]
rename_existing: M141.123
gcode:
    {% set s = params.S|default(0)|float %}
    SET_TEMPERATURE_FAN_TARGET temperature_fan="chamber" target={s}

```

**Reply by bluemorphe** [reply]
the renamed if exists is used to rename the old gcode, is that correct? and I put it in the file where we put the variables?
